src/Persistent_cohomology/include/gudhi/Persistent_cohomology.h

```diff
@@ -726,7 +727,7 @@ class Persistent_cohomology {
   /*  Dictionary establishing the correspondence between the Simplex_key of
    * the root vertex in the union-find ds and the Simplex_key of the vertex which
    * created the connected component as a 0-dimension homology feature.*/
-  std::map<Simplex_key, Simplex_key> zero_cocycles_;
+  std::unordered_map<Simplex_key, Simplex_key> zero_cocycles_;
```

Comment:
I imagine you also benchmarked this modification with the Simplex_tree ?

Reply:
I think so, but that was 3 months ago so I don't remember... I'll try again.

Reply:
In the branch `persline` (so I can use the benchmark line.cpp), I get for replacing map with unordered_map

**Before**
```
Mostly unique values
Cubical complex from top cells: 0.86s

Cubical complex from vertices: 0.746s

Simplex tree: 0.921s

Special case for function on a line: 0.007s

Many repeated values
Cubical complex from top cells: 1.043s

Cubical complex from vertices: 0.982s

Simplex tree: 0.628s

Special case for function on a line: 0.007s
```
**After**
```
Mostly unique values
Cubical complex from top cells: 0.702s

Cubical complex from vertices: 0.633s

Simplex tree: 0.772s

Special case for function on a line: 0.007s

Many repeated values
Cubical complex from top cells: 0.989s

Cubical complex from vertices: 0.934s

Simplex tree: 0.622s

Special case for function on a line: 0.007s
```
So sometimes is helps, sometimes it is neutral. I also tested on a Rips graph with a much larger ratio nb_edges / nb_vertices, and it decreased the time from 240ms to 200ms.